gcc/rust/rust-session-manager.h

```diff
@@ -205,9 +208,31 @@ struct CompileOptions
     enable_dump_option (DumpOption::TYPE_RESOLUTION_DUMP);
   }
 
+  /* Validate the crate name using the ASCII rules
+   TODO: Support Unicode version of the rules */
+  bool validate_crate_name (const std::string crate_name)
+  {
+    if (crate_name.length () > kMaxNameLength || crate_name.empty ())
+      return false;
+    for (auto &c : crate_name)
+      {
+	if (!(ISALNUM (c) || c == '_' || c == '-'))
+	  return false;
+      }
+    return true;
+  }
```

Comment:
This is something we can easily add unit tests for! If you think it would be useful, it would be great to have a base in place for when we implement UTF-8 crate names as you mention in the TODO above

Reply:
> This is something we can easily add unit tests for! If you think it would be useful, it would be great to have a base in place for when we implement UTF-8 crate names as you mention in the TODO above

I will add some unit-tests for this.

Reply:
Unit-tests added. But I am not sure if I am doing this right, since the only unit test framework I could find was the `selftest` module.

Reply:
selftest is the correct framework to use :) Good job